src/createAnimatedComponent.js

```diff
@@ -228,15 +236,31 @@ export default function createAnimatedComponent(Component) {
         ? this.props.style
         : [this.props.style];
       styles = flattenArray(styles);
-      const viewTag = findNodeHandle(this);
+
+      const hostInstance = ReactNative.findHostInstance_DEPRECATED(this);
+      // we can access view tag in the same way it's accessed here https://github.com/facebook/react/blob/e3f4eb7272d4ca0ee49f27577156b57eeb07cf73/packages/react-native-renderer/src/ReactFabric.js#L146
+      const viewTag = hostInstance._nativeTag;
+      /**
+       * RN uses viewConfig for components for storing different properties of the component(example: https://github.com/facebook/react-native/blob/master/Libraries/Components/ScrollView/ScrollViewViewConfig.js#L16).
+       * The name we're looking for is in the field named uiViewClassName.
+       */
+      const viewName = hostInstance.viewConfig.uiViewClassName;
+      // update UI props whitelist for this view
+      this._updateWhitelistProps(hostInstance);
```

Comment:
I think it'd be better for this to be moved outside of `createAnimatedComponent` class. No reason for it to be a class member method + now we also need to expose some implementation details from ConfigHelper which we do by making a copy each time the list of props is accessed.

I think it'd be better to move the implementation of this method directly to COnfigHelper. We add a new method there that is called `adaptViewConfig(viewConfig)` that gets the whole viewConfig such that it can use the className to remember which views have been processed and which not and also access the list of attributes.

We should also use Set() for whitelistViewNamesUpdated instead of an array as we do lookups which in case of an array are linear.

Reply:
As this method has side-effects can we make sure to only call it for reanimated 2 components? Otherwise we are changing behavior for people who are not using reanimated 2 and this is possibly a breaking change.

Reply:
fixed in 0cd1b52 and 8506102